recipes/protobuf/all/test_package/conanfile.py

```diff
@@ -1,24 +1,31 @@
 from conan import ConanFile
-from conan.tools.build import cross_building
-from conans import CMake
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake, CMakeToolchain
 import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
 
-    def build_requirements(self):
-        if hasattr(self, "settings_build") and cross_building(self):
-            self.build_requires(str(self.requires["protobuf"]))
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["protobuf_LITE"] = self.dependencies[self.tested_reference_str].options.lite
```

Comment:
doesn't `self.tested_reference_str` contain version as well, is it suitable for `self.dependencies`?

Reply:
it works